substrate/runtime/primitives/src/traits.rs

```diff
@@ -434,3 +427,46 @@ pub trait Applyable: Sized + Send + Sync {
 	fn sender(&self) -> &Self::AccountId;
 	fn apply(self) -> Result<(), &'static str>;
 }
+
+/// Something that acts like a `Digest` - it can have `Log`s `push`ed onto it and these `Log`s are
+/// each `Codec`.
+pub trait Digest {
+	type Item: DigestItem;
+
+	fn push(&mut self, item: Self::Item);
+}
+
+/// Single digest item.
+pub trait DigestItem: Member {
```

Comment:
Consider copying the `Event` infrastructure for this - it should be able to be used almost wholesale.